resources/widgetlist.ts

```diff
@@ -1,17 +1,38 @@
 // NOTE
 // This class should be considered deprecated, and any users of this class should
 // just switch over to using CSS grid.
+type Sorter = () => number;
 
 class WidgetList extends HTMLElement {
+  private _nextId!: number;
+  private _nameToId!: { [key: string]: number };
```

Comment:
Can `[key:string]` be more precise?
And why is here declared using `!.` oprator.